Saule/Serialization/ResourceSerializer.cs

```diff
@@ -235,24 +248,39 @@ private JObject SerializeRelationships(ResourceGraphNode node)
 
             foreach (var kv in node.Relationships)
             {
+                var relationship = kv.Value.Relationship;
+
                 var item = new JObject();
 
                 var data = SerializeRelationshipData(kv.Value);
 
                 var relationshipId = default(string);
 
                 if (data != null
-                    && kv.Value.Relationship.Kind == RelationshipKind.BelongsTo
+                    && relationship.Kind == RelationshipKind.BelongsTo
                     && kv.Value.SourceObject != null)
                 {
                     relationshipId = (string)data["id"];
                 }
 
                 var links = new JObject();
-                AddUrl(links, "self", _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), kv.Value.Relationship));
-                AddUrl(links, "related", _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), kv.Value.Relationship, relationshipId));
+                var self = _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), relationship);
+                var related = _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), relationship, relationshipId);
 
-                item["links"] = links;
+                if (!string.IsNullOrEmpty(self) && relationship.LinkType.HasFlag(LinkType.Self))
```

Comment:
I'm leaning on keeping it simple. The more complex path still hurts my brain. If one has other needs it could be implemented as a custom UrlBuilder. That said I might be shooting myself in the foot here. What is your opinion?